### ISSUE TITLE
Redid Support?

### ISSUE THREAD

**Issue by joshrathke** [onury/accesscontrol#49]
I’m looking to implement authorization in a clustered Node app. Has there been an attempt to make this package have Redis support? Or is there an alternative method of sharing memory between Node instances?

**Comment by aggarwaldev** [comment]
@onury Any Comments on this one?

**Comment by onury** [comment]
Yes, we should add this feature. Though, there are higher priority features we need to implement first.

**Comment by aggarwaldev** [comment]
@onury I think you should use "Feature Request" label instead of "Help Wanted".
Also, I think it's really important feature for clustered node applications. 

In the meantime, do you think a workaround would be possible to update roles when changes are made in database?

**Comment by onury** [comment]
"help wanted" stands for: contributors / collaborators are needed to implement this; and PRs are welcome. It does **not** indicate "this user is asking for help".

This is a good but extra feature. Unfortunately, AC itself cannot provide a "workaround" for this. 

What you can do for a workaround is; either have your app instances periodically check (poll) for updates; or horizontally update or notify instances of your service/application; at the time when any policy is changed. Depends on the architecture of your application. e.g. if using sockets with a central server, you could even do subscriptions.

**Comment by mikesparr** [comment]
I've been researching how to put it all together in this repo (here is authorization helper with Redis [cache] functionality I'm using if it helps): 

 * https://github.com/mikesparr/typescript-postgres-auth-example/blob/master/src/utils/authorization.helper.ts

**Comment by onury** [comment]
I'll check it soon. Thanks.

**Comment by mikesparr** [comment]
Thank YOU for putting together a great library. I just wrapped up integration tests and 51 tests green. Enjoy the repo: https://github.com/mikesparr/typescript-postgres-auth-example

**Comment by onury** [comment]
AccessControl deliberately stays out of transport/storage, but it gives you everything to share one model across a cluster: it serializes to plain JSON via `snapshot()` and rebuilds with `restore()`. Keep one source of truth, put the snapshot in Redis, and `restore()` it in each instance (re-pull on change, or via pub/sub):

```js
await redis.set('ac:policy', JSON.stringify(ac.snapshot()));
const ac = new AccessControl().restore(JSON.parse(await redis.get('ac:policy')));
```

The check itself runs locally in-memory in each instance — no per-request Redis round-trip. See the [one-policy-across-services recipe](https://onury.io/accesscontrol/guides/recipes/). Closing, as Redis/transport is an application concern rather than a library feature.

Sorry for the late reply!
